### PR TITLE
fix: fix predicate_index and not matched null cast

### PR DESCRIPTION
**Pull request by JackTan25**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
1.  fix not match null cast
2. fix predicate index error
Summary about this PR

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13208)

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_merge_into.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@
 use std::collections::HashMap;
 use std::sync::Arc;
 use std::time::Instant;
+use std::usize::MAX;
```
Comment (ariesdevil) [inline src/query/service/src/interpreters/interpreter_merge_into.rs:18]: better to use determined integer type: u32 or u64

File: src/query/service/src/interpreters/interpreter_merge_into.rs [hunk 2]
```diff
@@ -51,6 +53,8 @@ use crate::pipelines::PipelineBuildResult;
 use crate::schedulers::build_query_pipeline_without_render_result_set;
 use crate::sessions::QueryContext;
 
+// predicate_index should not be conflict with update expr's column_binding's index.
+pub const PREDICATE_COLUMN_INDEX: IndexType = MAX;
```
Comment (ariesdevil) [inline src/query/service/src/interpreters/interpreter_merge_into.rs:57]: does this conflict with https://github.com/datafuselabs/databend/blob/0afc3f3eb52daea8794bca1fe9b1506a6a8bed37/src/query/expression/src/schema.rs#L49

Comment (JackTan25) [reply src/query/service/src/interpreters/interpreter_merge_into.rs:57]: no, it's just a tag to find column. it's use for columnBounding's index, please just this think it's a name.

File: src/query/service/src/interpreters/interpreter_merge_into.rs (hunk 1 above)
Comment (JackTan25) [reply src/query/service/src/interpreters/interpreter_merge_into.rs:18]: ok

**Review by ariesdevil — APPROVED** [review]

File: src/query/service/src/interpreters/interpreter_merge_into.rs (hunk 2 above)
Comment (JackTan25) [reply src/query/service/src/interpreters/interpreter_merge_into.rs]: @ariesdevil wait, it's riskful.

Comment (JackTan25) [reply src/query/service/src/interpreters/interpreter_merge_into.rs]: it's safe, in metadata, the row_id_col idx is retrived by metatda.columns.len().

**Comment by dantengsky** [comment]
askbend:summary

**Review by dantengsky — APPROVED** [review]

**Pull request merged**